src/EnergyPlus/UnitarySystem.cc

```diff
@@ -3521,7 +3555,7 @@ namespace UnitarySystems {
             }
 
             // check if the UnitarySystem is connected to an outside air system
-            if (!AirLoopFound && state.dataSize->CurOASysNum > 0) {
+            if (!AirLoopFound && state.dataAirLoop->NumOASystems > 0) {
```

Comment:
If we are going to look through the OA system components then it does not matter if this model is called from the OA system or not, the end result is if the UnitarySystem resides there, not that it was called from there.